sys/include/net/gnrc/tcp.h

```diff
@@ -47,10 +47,26 @@ int gnrc_tcp_init(void);
 /**
  * @brief Initialize Transmission Control Block (TCB)
  * @pre @p tcb must not be NULL.
+ * @pre @p address_family must not be NULL.
+ * @pre @p local_addr must not be NULL.
+ * @pre @p local_port must not be NULL.
+ * @pre @p local_addr must not be NULL.
+ * @pre @p local_port must not be NULL.
+ * @pre @p ll_iface must not be NULL.
  *
  * @param[in,out] tcb   TCB that should be initialized.
+ * @param[in]     address_family    Pointer to address family storage.
+ * @param[in]     local_addr        Pointer to local address storage.
+ * @param[in]     local_addr_size   size of memory behind @p local_addr.
+ * @param[in]     local_port        Pointer to local port number storage.
```

Comment:
Since all address information shall be stored and supplied via sock_ep_t later on, I don't see any point in reimplementing a struct that does the same for gnrc_tcp.

Reply:
Then use a struct that maps to `sock_tcp_ep_t`. E.g.

```c
typedef struct {
    int family;
    ipv6_addr_t ipv6;
    uint16_t netif;
    uint16_t port;
} gnrc_tcp_ep_t;
```

However, be aware of the discussion going on in #12680.

As I said below: this API should be able to stand on its own. Furthermore, if you want to a struct anyway outside, you add further memory baggage by making each parameter its own argument. Up to four arguments can be stored in registers (on ARM) anything above costs stack memory. The current approach seems like the least ideal to me.

Reply:
I though about providing a binary compatible struct to sockt_ep_t but if I do that, I have to somehow detect changes in sock_ep_t (comparing struct sizes might not be enough), that is why I though providing raw pointers is the better approach. 

The memory consumption argument is valid, but the only changed function is gnrc_tcp_init, which is called only once at the start of a program so increased memory consumption does not really apply in this case.

From an Design Point of view providing a dedicated struct is a better approach. While I am at it, it would make sence to rewrite parts of gnrc_tcp_active_open and gnrc_tcp_passive_open. Both operate on raw pointers and could make use of a dedicated struct. 

What do you think?

Reply:
I just realized that a dedicated struck does not need to be binary compatible. I think I'll know how to improve this PR.